client.lua

```diff
@@ -1,25 +1,39 @@
 CreateThread(function()
+	local playerPed = PlayerPedId()
+	local boneId = 31086
+
 	RequestStreamedTextureDict("uhhh")
+
+	while not HasStreamedTextureDictLoaded("uhh")
+		Wait(0)
+	end
 	
-	while true do Wait(0)
-		local ratio = GetAspectRatio()
-		
+	while true do
 		local pool = GetGamePool("CPed")
-		table.sort(pool, function(a, b) return #(GetEntityCoords(a) - GetFinalRenderedCamCoord()) > #(GetEntityCoords(b) - GetFinalRenderedCamCoord()) end)
+		local camCoord = GetFinalRenderedCamCoord()
+
+		table.sort(pool, function(a, b) return #(GetEntityCoords(a) - camCoord) > #(GetEntityCoords(b) - camCoord) end)
 		
 		for i,v in pairs(pool) do
-			local boneId = 31086
+			playerPed = PlayerPedId()
+
 			local pos = GetWorldPositionOfEntityBone(v, GetPedBoneIndex(v, boneId))
-			local dist = #(GetFinalRenderedCamCoord() - pos)
-			local scale = (1.0/dist) * (1.0/GetFinalRenderedCamFov()) * 12.0
+			local dist = #(camCoord - pos)
 			local visible = IsEntityVisible(v)
-			local los = HasEntityClearLosToEntity(PlayerPedId(), v, 17) or v == PlayerPedId()
+			local los = HasEntityClearLosToEntity(playerPed, v, 17) or v == playerPed
 			
 			if dist < 50.0 and los and visible then
+				local ratio = GetAspectRatio()

```

Comment:
a) why put this inside the if statement?
b) why do you not get this once, like the playerped? this one is much more likely to remain the same throughout a session.

Reply:
a) we dont use ratio variable before dist, los and visible check so there is no reason to create local and call that native
b) but still someone can change thier aspect ratio whille playing